olive/passes/olive_pass.py

```diff
@@ -315,4 +342,13 @@ def validate_config(cls, v, values):
         return validate_config(v, PassConfigBase, config_class)
 
     def create_pass(self):
-        return Pass.registry[self.type.lower()](self.config, self.disable_search)
+        pass_cls = Pass.registry[self.type.lower()]
+        return create_pass_from_dict(pass_cls, self.config, self.disable_search)
+
+
+def create_pass_from_dict(pass_cls: Type[Pass], config: Dict[str, Any] = None, disable_search=False) -> Pass:
```

Comment:
can we also accept `str` as type for `pass_cls`? If it is a string, we could do ` Pass.registry[pass_cls.lower()]` to get the actual class. This might make it easier for users since they don't have to import the class.

Reply:
I'd like to move the logic to lookup type from string out of this function to make sure it only do one thing.

Reply:
I see. That sounds good to me. Thanks! 

This will only be used by olive workflow or advanced user, so my request is not that important for user experience.